File: ZoomingPDFViewer/ZoomingPDFViewerCocoa/ZoomingPDFViewer/DataViewController.m
/*
Copyright (C) 2017 Apple Inc. All Rights Reserved.
See LICENSE.txt for this sample’s licensing information

Abstract:
The app's view controller which presents viewable content.
*/


#import "DataViewController.h"
#import "PDFView.h"
#import "TiledPDFScrollView.h"
#import "TiledPDFView.h"



@implementation DataViewController



- (void)dealloc
{
    if ( self.page != NULL )
    {
        CGPDFPageRelease( self.page );
    }
}



- (void)viewDidLoad
{
    [super viewDidLoad];

	// Do any additional setup after loading the view, typically from a nib.
    self.page = CGPDFDocumentGetPage( self.pdf, self.pageNumber );
    if ( self.page != NULL ) CGPDFPageRetain( self.page );
    [self.scrollView setPDFPage:self.page];

    // Disable zooming if our pages are currently shown in landscape for new views
    [self.scrollView setUserInteractionEnabled:UIInterfaceOrientationIsPortrait([UIApplication sharedApplication].statusBarOrientation)];
}



- (void)viewDidLayoutSubviews
{
    [self restoreScale];
}



- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id <UIViewControllerTransitionCoordinator>)coordinator
{
    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];

    [coordinator
     animateAlongsideTransition:nil
     completion:^(id<UIViewControllerTransitionCoordinatorContext> context) {

         // Disable zooming if our pages are currently shown in landscape after orientation changes
         [self.scrollView setUserInteractionEnabled:UIInterfaceOrientationIsPortrait([UIApplication sharedApplication].statusBarOrientation)];
    }];
}



- (void)restoreScale
{
    // Called on orientation change.
    // We need to zoom out and basically reset the scrollview to look right in two-page spline view.
    CGRect pageRect = CGPDFPageGetBoxRect( self.page, kCGPDFMediaBox );
    CGFloat yScale = self.view.frame.size.height / pageRect.size.height;
    CGFloat xScale = self.view.frame.size.width / pageRect.size.width;
    self.myScale = MIN( xScale, yScale );
    self.scrollView.bounds = self.view.bounds;
    self.scrollView.zoomScale = 1.0;
    self.scrollView.PDFScale = self.myScale;
    self.scrollView.tiledPDFView.bounds = self.view.bounds;
    self.scrollView.tiledPDFView.myScale = self.myScale;
    [self.scrollView.tiledPDFView.layer setNeedsDisplay];
}


@end

File: ZoomingPDFViewer/ZoomingPDFViewerSwift/ZoomingPDFViewer/DataViewController.swift
/*
Copyright (C) 2017 Apple Inc. All Rights Reserved.
See LICENSE.txt for this sample’s licensing information

Abstract:
The app's view controller which presents viewable content.
*/


import UIKit

class DataViewController: UIViewController {


    @IBOutlet weak var scrollView: TiledPDFScrollView!

    var pdf: CGPDFDocument!

    var page: CGPDFPage!

    var pageNumber: Int = 0
    
    var myScale: CGFloat = 0



    override func viewDidLoad()
    {
        super.viewDidLoad()

        // Do any additional setup after loading the view, typically from a nib.
        page = pdf.page(at: pageNumber)
        scrollView.setPDFPage(page)

        // Disable zooming if our pages are currently shown in landscape, for new views
        scrollView.isUserInteractionEnabled = UIApplication.shared.statusBarOrientation.isPortrait
    }



    override func viewDidLayoutSubviews()
    {
        restoreScale()
    }



    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator)
    {
        super.viewWillTransition(to: size, with: coordinator)

        coordinator.animate(alongsideTransition: nil, completion: { context in

            // Disable zooming if our pages are currently shown in landscape after orientation changes
            self.scrollView.isUserInteractionEnabled = UIApplication.shared.statusBarOrientation.isPortrait
        })
    }



    func restoreScale()
    {
        // Called on orientation change.
        // We need to zoom out and basically reset the scrollview to look right in two-page spline view.
        let pageRect = page.getBoxRect(CGPDFBox.mediaBox)
        let yScale = view.frame.size.height / pageRect.size.height
        let xScale = view.frame.size.width / pageRect.size.width
        myScale = min(xScale, yScale)
        scrollView.bounds = view.bounds
        scrollView.zoomScale = 1.0
        scrollView.PDFScale = myScale
        scrollView.tiledPDFView.bounds = view.bounds
        scrollView.tiledPDFView.myScale = myScale
        scrollView.tiledPDFView.layer.setNeedsDisplay()
    }


}

